src/Sfa.Das.Sas.Infrastructure/Provider/ElasticSearch/ElasticsearchProviderIndexMaintainer.cs

```diff
@@ -50,6 +50,19 @@ public override void CreateIndex(string indexName)
             }
         }
 
+        public override bool IndexContainsDocuments(string indexName)
+        {
+            var providerDocuments = Client.Search<ProviderDocument>(s => s.Index(indexName).Size(0).MatchAll());
+            var providerApiDocuments = Client.Search<ProviderApiDocument>(s => s.Index(indexName).Size(0).MatchAll());
+            var standardProviderDocuments = Client.Search<StandardProvider>(s => s.Index(indexName).Size(0).MatchAll());
+            var frameworkProviderDocuments = Client.Search<FrameworkProvider>(s => s.Index(indexName).Size(0).MatchAll());
+
+            return providerDocuments.HitsMetaData.Total > 0
+                   && providerApiDocuments.HitsMetaData.Total > 0
+                   && standardProviderDocuments.HitsMetaData.Total > 10000
+                   && frameworkProviderDocuments.HitsMetaData.Total > 100000;
+        }
+
```

Comment:
looks good to me and I won't defer approval on this point, but do we still care about providerDocuments anyway?

Reply:
We do until we have a story about removing them from the index